client/admin/chat-channels.tsx

```diff
@@ -0,0 +1,269 @@
+  height: 88px;
+  margin: 16px 0;
+`
+
+const NoResults = styled.div`
+  ${subtitle1};
+
+  color: ${colorTextFaint};
+`
+
+const ErrorText = styled.div`
+  ${subtitle1};
+
+  color: ${colorError};
+`
+
+const StyledCarousel = styled(Carousel)`
+  max-width: 960px;
+`
+
+const ChannelThumbnail = styled(Card)`
+  width: 420px;
+  padding: 16px;
+
+  flex-shrink: 0;
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+
+  &:not(:first-child) {
+    margin-left: 16px;
+  }
+`
+
+const StyledChannelIcon = styled(ChannelIcon)`
+  width: 56px;
+  height: auto;
+  flex-shrink: 0;
+`
+
+const ChannelInfoContainer = styled.div`
+  flex-grow: 1;
+  display: flex;
+  flex-direction: column;
+  margin: 0 8px;
+  overflow: hidden;
+`
+
+const ChannelInfoName = styled(Headline6)`
+  ${singleLine};
+`
+
+const ChannelHeadline = styled.div`
+  ${headline6};
+  margin-bottom: 8px;
+`
+
+const ChannelContainer = styled.div`
+  width: 100%;
+  overflow: hidden;
+
+  flex-grow: 1;
+  display: flex;
+`
+
+const StyledMessageList = styled(MessageList)`
+  flex-grow: 1;
+  max-width: 960px;
+  min-width: 320px;
+`
+
+export function AdminChatChannels() {
+  const dispatch = useAppDispatch()
+  const [channels, setChannels] = useState<ChannelInfo[]>()
+  const [currentPage, setCurrentPage] = useState(-1)
+  const [hasMoreChannels, setHasMoreChannels] = useState(true)
+
+  const [isLoadingMoreChannels, setIsLoadingMoreChannels] = useState(false)
+  const [searchError, setSearchError] = useState<Error>()
+  const [searchQuery, setSearchQuery] = useLocationSearchParam('q')
+
+  const carouselRef = useRef(null)
+  const debouncedSearchRef = useRef(
+    debounce((searchQuery: string) => {
+      // Just need to clear the search results here and let the carousel initiate the network
+      // request.
+      setSearchQuery(searchQuery)
+      setSearchError(undefined)
+      setChannels(undefined)
+      setCurrentPage(-1)
+      setHasMoreChannels(true)
+      if (carouselRef.current) {
+        // TODO(2Pac): Fix the types once the carousel is ported to TypeScript.
+        ;(carouselRef.current as any).reset()
+      }
+    }, 100),
+  )
+
+  const onSearchChange = useStableCallback((searchQuery: string) => {
+    debouncedSearchRef.current(searchQuery)
+  })
+
+  const onLoadMoreChannels = useStableCallback(() => {
+    setIsLoadingMoreChannels(true)
+    setCurrentPage(currentPage + 1)
+
+    fetchJson<ChannelInfo[]>(
+      apiUrl`admin/chat/?q=${searchQuery}&limit=${SEARCH_CHANNELS_LIMIT}&page=${currentPage + 1}`,
+    )
+      .then(data => {
+        setIsLoadingMoreChannels(false)
+        setChannels((channels ?? []).concat(data))
+        setHasMoreChannels(data.length >= SEARCH_CHANNELS_LIMIT)
+      })
+      .catch(err => {
+        setIsLoadingMoreChannels(false)
+        setSearchError(err)
+      })
+  })
+
+  const [selectedChannel, setSelectedChannel] = useState<ChannelInfo>()
+  const channelUsers = useAppSelector(s =>
```

Comment:
imo it'd be a lot better to split this stuff out into a component you use here and in the actual channel, this feels like way too much duplication